torch/_subclasses/fake_tensor.py

```diff
@@ -821,40 +821,30 @@ def __torch_dispatch__(self, func, types, args=(), kwargs=None):
         # is written to must be invalidated
         self.invalidate_written_to_constants(func, flat_arg_fake_tensors, args, kwargs)
 
-        from torch._decomp import decomposition_table
-
-        with self:
-            # Decomposes CompositeImplicitAutograd ops
-            r = func.decompose(*args, **kwargs)
-            if r is not NotImplemented:
-                return r
+        # If there's a Python meta, prefer that over the decomposition
+        from torch._decomp import meta_table as meta_table
 
-        # IDK: feels bad man, sym_numel on as_strided infinite loops otherwise
-        if has_symbolic_sizes and not self.cpp_meta_supports_symint(func):
-            from torch._decomp import meta_table as meta_table
+        if func not in meta_table:
+            from torch._decomp import decomposition_table
 
-            if func == aten.size.default:
-                sys.stderr.write(
-                    "Trying to call aten.size on a tensor with symbolic shapes. "
-                    "It's likely that this is from calling tensor.shape in C++"
+            # Prefer Python decompositions over C++ ones
+            if func in decomposition_table and (
+                has_symbolic_sizes
+                or (
+                    # TODO: Remove these exclusions, so that we can remove
+                    # this leg entirely
+                    torch_decomp_decompositions(func)
```

Comment:
What remains to be done in order to remove this torch_decomp_decompositions() check?

Reply:
I need to remove it and see what fails